pkgs/development/libraries/mbedtls/default.nix

```diff
@@ -10,47 +19,19 @@ stdenv.mkDerivation rec {
     sha256 = "0im83kqf7a64ywxh6dnv0by3gwxww93zx5wpdqglr6xp7b8yg4xk";
   };
 
-  nativeBuildInputs = [ perl ];
-
-  postPatch = ''
-    patchShebangs .
-  '' + stdenv.lib.optionalString stdenv.isDarwin ''
-    substituteInPlace library/Makefile --replace "-soname" "-install_name"
-    substituteInPlace tests/scripts/run-test-suites.pl --replace "LD_LIBRARY_PATH" "DYLD_LIBRARY_PATH"
-    # Necessary for install_name_tool below
-    echo "LOCAL_LDFLAGS += -headerpad_max_install_names" >> programs/Makefile
-  '';
+  nativeBuildInputs = [ cmake ninja perl ];
 
-  makeFlags = [
-    "SHARED=1"
-  ] ++ stdenv.lib.optionals stdenv.isDarwin [
-    "DLEXT=dylib"
-  ];
-
-  installFlags = [
-    "DESTDIR=\${out}"
-  ];
-
-  postInstall = stdenv.lib.optionalString stdenv.isDarwin ''
-    install_name_tool -change libmbedcrypto.dylib $out/lib/libmbedcrypto.dylib $out/lib/libmbedtls.dylib
```

Comment:
Possibly ask the author of these lines.

Reply:
@Mic92 Good day!

Long story short.
I migrate mbed TLS to officially supported CMake.

Mostly Linux and Mac builds seems to be clean.

Please say your toughs on these changes that undo also `positInstalls` from https://about.gitlab.com/2016/06/22/gitlab-8-9-released/#import--export-projects.

I can not find way to text macOS, or get a bundle of builded by bot `/nix/store/csyz7nid4zkf766z9djsxspx4b6cafnd-mbedtls-2.10.0` to investigate.

So, please say your thoughts on these changes.

Reply:
I think the fixes are there primarily for the consumers of the library

Reply:
I did not add these lines. 
Are those required? @mnacamura

Reply:
My addition was the `DLEXT=dylib` part only. If extension is `.so`, some programs cannot find library on darwin. So I added it.

The contents of `postInstall` was I think added in  #20940.

Reply:
Summary: We decided that this clean-up of code is legitimate, so it stays.